pkg/controller/garbagecollector/garbagecollector_test.go

```diff
@@ -2462,6 +2520,34 @@ func makeDeleteEvent(identity objectReference, owners ...objectReference) *event
 	}
 }
 
+func makeUpdateEvent(identity objectReference, owners ...objectReference) *event {
+	gv, err := schema.ParseGroupVersion(identity.APIVersion)
+	if err != nil {
+		panic(err)
+	}
+	return &event{
+		eventType: updateEvent,
+		gvk:       gv.WithKind(identity.Kind),
+		obj:       makeObj(identity, owners...),
```

Comment:
hmm... seems like we have to set oldObj in the event as well (and delete the godoc that says "it's not used by the garbage collector", because it definitely is used by the processTransitions logic)